src/events/embedded/types.ts

```diff
@@ -1,116 +1,81 @@
-export interface IEmbeddedMessageElementsButton {
+interface Session {
   id: string;
-  title: string;
-  action?: IEmbeddedMessageElementsButtonAction;
+  start?: number;
+  end?: number;
 }
-export interface IEmbeddedMessageElementsButtonAction {
+
+interface Impression {
+  messageId: string;
+  displayCount: number;
+  displayDuration: number;
+}
+
+interface ButtonAction {
   type: string;
   data?: string;
 }
-export interface IEmbeddedMessageElements {
+
+interface Button {
+  id: string;
+  title: string;
+  action?: ButtonAction;
+}
+
+interface Elements {
   title?: string;
   body?: string;
   mediaUrl?: string;
 
-  buttons?: [IEmbeddedMessageElementsButton];
-  text?: [IEmbeddedMessageElementsText];
-  defaultAction?: IEmbeddedMessageElementsDefaultAction;
+  buttons?: Button[];
+  text?: Text[];
+  defaultAction?: DefaultAction;
 }
-export interface IEmbeddedMessageElementsText {
+
+interface Text {
   id: string;
   text?: string;
 }
-export interface IEmbeddedMessageElementsDefaultAction {
+
+interface DefaultAction {
   type: string;
   data?: string;
 }
 
-export interface IEmbeddedMessageMetadata {
+interface Metadata {
   messageId: string;
   campaignId?: number;
   isProof?: boolean;
   placementId?: number;
 }
 
-export interface DeviceInfo {
-  deviceId: string;
-  platform: string;
-  appPackageName: string;
-}
-
-export interface IEmbeddedMessageData {
-  email?: string;
-  userId?: string;
-  messageId: string;
-  metadata: IEmbeddedMessageMetadata;
-  elements?: IEmbeddedMessageElements;
-  payload?: Array<any>;
-}
-
-export interface IEmbeddedMessage {
+export interface EmbeddedMessage {
   email?: string;
   userId?: string;
   messageId: string;
-  metadata?: IEmbeddedMessageMetadata;
-  elements?: IEmbeddedMessageElements;
+  metadata: Metadata;
+  elements?: Elements;
   payload?: Array<any>;
```

Comment:
Is Array<any> the best type to use here? I think a tighter type for embedded payload would be Record<string, any>